gpu_bdb/bdb_tools/q04_utils.py

```diff
@@ -58,8 +60,8 @@ def abandonedShoppingCarts(df, DYNAMIC_CAT_CODE, ORDER_CAT_CODE):
     # (eg:123456:3, 234567:5)
     filtered_df["wp_type_codes"] = (
         filtered_df["tstamp_inSec"]
-        .astype("str")
-        .str.cat(filtered_df["wp_type_codes"].astype("str"), sep=":")
+        .astype(str)
+        .str.cat(filtered_df["wp_type_codes"].astype(str), sep=":")
```

Comment:
I suggesting using `string` to follow the recommended practice according to pandas docs https://pandas.pydata.org/pandas-docs/stable/user_guide/text.html . 

```suggestion
        .astype("string")
        .str.cat(filtered_df["wp_type_codes"].astype("string"), sep=":")
```